cli/snapshot.go

```diff
@@ -0,0 +1,169 @@
+package cli
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/signal"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/urfave/cli"
+	"golang.org/x/sys/unix"
+)
+
+const (
+	// ioctl magics from <linux/fs.h>
+	IOCTL_FIFREEZE_MAGIC = 0xC0045877 // _IOWR('X', 119, int)
+	IOCTL_FITHAW_MAGIC   = 0xC0045878 // _IOWR('X', 120, int)
+)
+
+type SnapshotOptions struct {
+	ArtifactName      string
+	ArtifactGroup     string
+	CompatibleDevices []string
+	Depends           []string
+	Provides          []string
+}
+
+type mountInfo struct {
+	mountID    int
+	mountPoint string
+	mountSrc   string
+	fsType     string
+}
+
+func (runOpts *runOptionsType) DumpSnapshot(ctx *cli.Context) error {
```

Comment:
As mentioned in [MEN-2931](https://tracker.mender.io/browse/MEN-2931), we might want to ensure that logging calls don't go to the stream anywhere in here. This is also for future modifications, a logging call can easily be added without knowing that this will go to the stream.

So I suggest calling `log.SetOutput(os.Stderr)` to switch it to standard error. Does this make sense?